metadata-ingestion/src/datahub/ingestion/source/s3/__init__.py

```diff
@@ -229,9 +229,11 @@ def init_spark(self):
 
         if self.source_config.aws_config is not None:
 
-            aws_access_key_id = self.source_config.aws_config.aws_access_key_id
-            aws_secret_access_key = self.source_config.aws_config.aws_secret_access_key
-            aws_session_token = self.source_config.aws_config.aws_session_token
+            credentials = self.source_config.aws_config.get_credentials()
```

Comment:
Assuming we want to continue using boto3 as directly as possible, we could directly call out to `self.source_config.aws_config.get_session().get_credentials()` here

Reply:
Gonna leave that up to a datahub maintainer to decide. There is a slight bit of logic in the `get_credentials()` function for `None` Checking

Reply:
I think the current pattern is fine!